cmake/app/boilerplate.cmake

```diff
@@ -525,6 +525,8 @@ if(CONFIG_QEMU_TARGET)
   endif()
 endif()
 
+add_custom_target(app_mem_partition)
+
```

Comment:
Can we rename the target to something generic so it can be reused for other properties without needing to define yet another target?

e.g.
target=zephyr_target
property=app_mem_partition_compile_options

Reply:
@SebastianBoe good suggestion.
The target could have a more generic name, `zephyr_target` is fine.
Unfortunately, the property used is defined by CMake: https://cmake.org/cmake/help/latest/manual/cmake-properties.7.html#properties-on-targets

Which makes it a bit trickier.
We could decide to use the property `COMPILE_OPTINS` for the app_memory, and then use e.g. `COMPILE_FEATURES`, `LINK_OPTIONS`, etc for other settings we need in future.
A bit misuse, but as we wrap the usage into `zephyr_library_<cmd>` users won't need to care, and we limit the number of targets :)

Other thoughts ?

Reply:
Actually, one can read out custom properties from custom targets. It's just the non-custom targets, like libraries, that can only have a fixed set of properties.

At least this worked fine when we did this here:
https://github.com/zephyrproject-rtos/zephyr/pull/13672/files#diff-b9537eb0f71fef9b0fdb1e934a042cecR551

Reply:
correct, but that won't work with generator expressions.
(unless latest CMake has changed that).
Only the properties that specifically states:
https://cmake.org/cmake/help/latest/prop_tgt/COMPILE_OPTIONS.html
```
Contents of COMPILE_OPTIONS may use “generator expressions” with the syntax $<...>.
```

The reason for using a generator expression in this case, is that we don't need to consider whether the file is named `lib<name>.a` or something else.
https://github.com/tejlmand/zephyr/blob/066af900261246f23ae3cdb687865f6d7ef4c8c2/cmake/extensions.cmake#L497
```
$<TARGET_FILE_NAME:${ZEPHYR_CURRENT_LIBRARY}>
```

Alternative is of course to simply do:
```
"lib${ZEPHYR_CURRENT_LIBRARY}.a"
```
but out of principle I dislike having to use such knowledge in the build system, as it assumes a bit too much on compiler and linker.

Reply:
I'm pretty sure it works with generator expressions, it could be the documentation is misleading.

See

https://github.com/NordicPlayground/fw-nrfconnect-nrf/blob/master/subsys/bootloader/image/CMakeLists.txt#L90

for instance.